### ISSUE TITLE
Landlab dev install PackagesNotFound

### ISSUE THREAD

**Issue by pfeiffea** [landlab/landlab#1704]
Hey team. I'm struggling doing a developer install (following instructions [here](https://landlab.readthedocs.io/en/master/install/index.html#)) on a new machine. My branch is up to date with landlab/master, conda is up to date. I set up a new landlab_dev environment. 

Issue 1 (resolved, but do instructions need an update?): 

`conda install --file=requirements.in` gives me the following error

```
PackagesNotFoundError: The following packages are not available from current channels:

  - bmipy
  - rich-click
 ```

Based on [1486](https://github.com/landlab/landlab/issues/1486), I tried running `conda install landlab -c conda-forge` first, and this solved the problem. _Should installing regular landlab be a step written in to the developer install documentation?_ 

Issue 2: 
`pip install -e .` gives me the following 

```
Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error

  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [94 lines of output]

      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          function be evaluated with x=1 to determine if it any erosion occured.

          """
          cdef double f

          f = x - 1.0 + (alpha * (x ** n)) - beta
                                           ^
      ------------------------------------------------------------

      landlab\components\stream_power\cfuncs.pyx:360:37: Cannot assign type 'npy_double complex' to 'double'
```
...
```
      Cython.Compiler.Errors.CompileError: landlab\components\stream_power\cfuncs.pyx
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

I remember that I've had cython/compiler errors before related to Windows machines and C++ compilers, but now I can't find those instructions (referenced in [1513](https://github.com/landlab/landlab/issues/1513), but link is broken) on the updated documentation pages. 

Any suggestions?

**Comment by mcflugen** [comment]
@pfeiffea I suspect you just need to add `-c conda-forge` to the install command.

**Comment by pfeiffea** [comment]
Hey @mcflugen - perfect timing, I was just updating my issue. Yes, `-c conda-forge` was definitely the solution to issue #1.

**Comment by mcflugen** [comment]
@pfeiffea I've not seen your second issue before. Have you installed a C and C++ compiler?
```
conda install c-compiler cxx-compiler
```
After doing this, you may have to deactivate your environment and then re-activate it.

**Comment by mcflugen** [comment]
> Should installing regular landlab be a step written in to the developer install documentation?

No, I don't think so. Just using the `-c conda-forge` option should be enough to install all of the dependencies. I will add that to the install instructions.

**Comment by mcflugen** [comment]
@pfeiffea What version of Microsoft Visual C++ do you have installed? You should be using 14.x.

**Comment by pfeiffea** [comment]
Thanks for the help with this @mcflugen 

I ran `conda install c-compiler cxx-compiler -c conda-forge` to run (note that I had to add -c conda-forge for it to run successfully). deactivated and then rectivated landlab_dev. After doing this I get the same set of errors when running `pip install -e .` 

In Apps-> Installed apps I have quite a few Visual C++ versions listed, including 14.36.32532. I'll note that I didn't go through a separate installation of a compiler like I remember doing on new machines in the past. 

I'm running Windows 11, in case that makes any difference.

**Comment by mcflugen** [comment]
@pfeiffea I see the problem. Your fork isn't synced with the latest version of *Landlab*. Have a look at the line in question with the latest version.
```cython
f = x - 1.0 + (alpha * (x ** n)) - beta
```
vs.
```cython
f = x - 1.0 + (alpha * pow(x, n)) - beta
```
In the first, `x ** n` can return a complex value (which is the source of the error) but `pow(x, n)` always returns a float.

**Comment by pfeiffea** [comment]
@mcflugen I owe you one- looks like I botched something while updating things yesterday. I did a hard reset of my fork and it fixed that problem. 

In order to get `pip install -e .` to run successfully I also had to download a Microsoft Visual C++ compiler (thankful for the clear error message there), which I find a bit confusing since there were some listed in my installed apps. Is that still an expected step for Windows users? If so, it would be great to add it back in to the Developer Install instructions.
